book.rb

```diff
@@ -0,0 +1,13 @@
+class Book
+  attr_accessor :title, :author, :rentals
+
+  def initialize(title, author)
+    @title = title
+    @author = author
+    @rentals = []
+  end
+
+  def add_rental(_book, person, date)
+    rentals << Rental.new(self, person, date)
+  end
```

Comment:
- [ Required ] I think it would be nice if you refactor this method for the following two reasons.
  
  - Since this is the book class you don't need to pass the `_book` object as a parameter in the method header, you can use the `self` object instead.
  
  - Since the rental object is being added to `book.rentals` in the `Rental` constructor you don't have to do it in the `add_rental` method.
  
 Please see the example code below for a suggestion of how you can achieve this.

## Example code

```ruby
require_relative 'rental'

class Book
  attr_accessor :title, :author, :rentals

  def initialize(title, author)
    @title = title
    @author = author
    @rentals = []
  end

  def add_rental(person, date)
    Rental.new(self, person, date)
  end
end
```

Reply:
Thank you for this amazing correction. I have implemented the changes.